localstack/services/cloudformation/models/iam.py

```diff
@@ -104,28 +165,95 @@ def update_resource(self, new_resource, stack_name, resources):
             RoleName=props.get("RoleName"), Description=props.get("Description") or ""
         )
 
-    @staticmethod
-    def get_deploy_templates():
+    @classmethod
+    def get_deploy_templates(cls):
+        def _post_create(resource_id, resources, resource_type, func, stack_name):
+            """attaches managed policies from the template to the role"""
+            iam = aws_stack.connect_to_service("iam")
+            resource = resources[resource_id]
+            props = resource["Properties"]
+            role_name = props["RoleName"]
+
+            # attach managed policies
+            policy_arns = props.get("ManagedPolicyArns", [])
+            for arn in policy_arns:
+                iam.attach_role_policy(RoleName=role_name, PolicyArn=arn)
+
+            # add inline policies
+            inline_policies = props.get("Policies", [])
+            for policy in inline_policies:
+                # TODO: when would this be a list?
+                policy = policy[0] if isinstance(policy, list) and len(policy) == 1 else policy
+                if policy == PLACEHOLDER_AWS_NO_VALUE:
+                    continue
+                if not isinstance(policy, dict):
+                    LOG.info(
+                        'Invalid format of policy for IAM role "%s": %s'
+                        % (props.get("RoleName"), policy)
+                    )
+                    continue
+                pol_name = policy.get("PolicyName")
+                doc = dict(policy["PolicyDocument"])
+                doc["Version"] = doc.get("Version") or IAM_POLICY_VERSION
+                statements = (
```

Comment:
nit: Unrelated to your changes - but could be replaced with `ensure_list(..)` (recently added to `common.py`):
```
                statements = ensure_list(doc["Statement"])
```